src/algorithms/tonal/crepe.cpp

```diff
@@ -0,0 +1,131 @@
+  "timestamp as value between 0 to 1. `activations` is a time by sigmoid activations matrix returned "
+  " by the neural network.\n"
+  "\n"
+  "See TensorflowPredictCREPE for details about the rest of parameters.\n"
+  "The recommended pipeline is as follows::\n"
+  "\n"
+  "  MonoLoader(sampleRate=16000) >> CREPE()\n"
+  "\n"
+  "Note: This algorithm does not make any check on the input model so it is "
+  "the user's responsibility to make sure it is a valid one.\n"
+  "\n"
+  "References:\n"
+  "\n"
+  "1. CREPE: A Convolutional Representation for Pitch Estimation. "
+  "Jong Wook Kim, Justin Salamon, Peter Li, Juan Pablo Bello. "
+  "Proceedings of the IEEE International Conference on Acoustics, Speech, and Signal "
+  "Processing (ICASSP), 2018.\n"
+  "\n"
+  "2. Original models and code at https://github.com/marl/crepe/\n"
+  "\n"
+  "3. Supported models at https://essentia.upf.edu/models/\n\n");
+
+
+void CREPE::configure() {
+  _tensorflowPredictCREPE->configure(INHERIT("graphFilename"),
+                                     INHERIT("input"),
+                                     INHERIT("output"),
+                                     INHERIT("hopSize"),
+                                     INHERIT("batchSize"));
+
+  _hopSize = parameter("hopSize").toFloat();
+  // _viterbi = parameter("viterbi").toBool();
+
+  for (int i = 0; i < _nPitches - 1; ++i) {
+    _centsMapping.push_back(_delta * i + _shift);
+  }
+  _centsMapping.push_back(_end + _shift);
+}
+
+
+void CREPE::compute() {
+  const vector<Real>& audio = _audio.get();
+  vector<Real>& time = _time.get();
+  vector<Real>& frequency = _frequency.get();
+  vector<Real>& confidence = _confidence.get();
+  vector<vector<Real> >& activations = _activations.get();
+
+  _tensorflowPredictCREPE->input("signal").set(audio);
+  _tensorflowPredictCREPE->output("predictions").set(activations);
+  _tensorflowPredictCREPE->compute();
+
+  int timestamps = activations.size();
+
+  time.resize(timestamps);
+  frequency.resize(timestamps);
+  confidence.resize(timestamps);
+  
+  frequency.assign(timestamps, 0);
+
+  // Activations to cents.
+  vector<Real> cents = toLocalAverageCents(activations);
+
+  for (int i = 0; i < timestamps; i++) {
+    // Get the timestamp of each pitch prediction.
+    time[i] = i * _hopSize / 1000.0;
+
+    // Get the confidences (max Activation per timestamp).
+    confidence[i] = activations[i][argmax(activations[i])];
+
+    // Cents to frequencies.
+    if (!isnan(cents[i])) {
+      frequency[i] = 10.0 * pow(2.0, (cents[i] / 1200.0));
+    }
+  }
+}
+
+vector<Real> CREPE::toLocalAverageCents(vector<vector<Real> > &activations) {
+  int timestamps = activations.size();
+  vector<Real> cents(timestamps, 0);
+
+  int center, start, end;
+  Real productSum, weightSum;
+  vector<Real> salience, toCents;
+  for (int i = 0; i < timestamps; i++) {
+    center = argmax(activations[i]);
```

Comment:
Nice observation!
Our argmax implementation relies on [max_element](https://www.cplusplus.com/reference/algorithm/max_element/), which as NumPy's [argmax](https://numpy.org/doc/stable/reference/generated/numpy.argmax.html) returns the first element in case of multiple candidates.
Then we should decide if matching the original implantation is enough, or if we should look for a method to choose a candidate if there is more than one.

I couldn't find any issue related to this in the [original repo](https://github.com/marl/crepe).

Reply:
We can be as consistent with the original implementation as possible. But may be it is worth mentioning this issue somewhere, at least in the code comments.